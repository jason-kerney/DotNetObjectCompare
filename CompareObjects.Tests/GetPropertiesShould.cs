using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Linq;

namespace CompareObjects.Tests
{
	[TestClass]
	[UseReporter(typeof(BeyondCompareReporter))]
	public class GetPropertiesShould
	{
		[TestMethod]
		public void GetOnlyReadablePublicPropertiesFromAnObject()
		{
			var tested = ObjectUnderTest.GetDefaultCase();

			List<string> skipped;

			var properties = tested.GetProperties(out skipped);

			Assert.IsTrue(!skipped.Any());
			Approvals.VerifyAll(properties);
		}

		[TestMethod]
		public void SkippIndexedItems()
		{
			var tested = IndexedObjectUnderTest.GetDefaultCase();

			List<string> skipped;

			tested.GetProperties(out skipped);

			Approvals.VerifyAll(skipped, "skipped");
		}
	}
}