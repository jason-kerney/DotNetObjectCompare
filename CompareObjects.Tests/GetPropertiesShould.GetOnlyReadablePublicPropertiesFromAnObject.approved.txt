HiddenGetProperty => Hidden get property
IntProperty => 5
ReadonlyProperty => False
StringProperty => string property
